experimental/QuadFormAndIsom/src/embeddings.jl

```diff
@@ -735,15 +799,19 @@ function _primitive_extensions_generic(
           if ext_type[1] == :equivariant && ext_type[2] == :plain
             # We need to see whether there exists an isometry of N which
             # stabilizes HN and agrees with fHM along the gluing phig.
-            reporb = _fitting_isometries(OqfN, HNinqN, phig, fHM, discrep, stabN, N, first)
+            reporb = _fitting_isometries(OqfN, HNinqN, phig, fHM, discrep, stabN, N, first_fitting_isometry)
           else
             reporb = QQMatrix[fN]
           end
 
           for b in reporb
             L, fL, graph = _overlattice(phig, HMinD, HNinD, fM, b; same_ambient)
 
-            (!isnothing(q) && genus(L) == G) || (is_even(L) == even) || continue
+            if !isnothing(q) && genus(L) != G
+              continue
+            elseif is_even(L) != even
+              continue
+            end
```

Comment:
Previously the test was as `(!isnothing(q) && genus(L) == G) || (is_even(L) == even) || continue` which was wrong because if `q  !== nothing` and `genus(L) != G` we should discard `L`, but we were keeping it if it was even (which did not make sense at all here)